### PR TITLE
Make single ticket embeds smaller

### PR DESCRIPTION
**Pull request by violine1101**
This PR changes some things about single ticket embeds, so that they are not as large as before.

* Screenshots are now always thumbnails
* Text of the ticket is displayed as short as possible

### CODE REVIEW

**Comment by NeunEinser** [comment]
If we want to actually translate code blocks, that would be possible too, but it really only makes sense if the restriction of `\n` is somehow tweaked.
```ts
description = description.replace(
	/{code(?::([A-Za-z#+]+)(?:\|.+?)*|.+?)?}((?:.|\n)+?){code}/gm,
	(_, lang: String, code: String) => 
		`\`\`\`${ lang?.toLowerCase().replace('c#', 'cs').replace('c++', 'cpp') ?? 'java' }\n${ code.trim() }\n\`\`\``
	
)
```
Without line restriction:
image

File: src/mentions/SingleMention.ts [hunk 1]
```diff
@@ -70,8 +70,36 @@ export class SingleMention extends Mention {
...
+
+		// remove panels
 		description = description.replace( /\s*\{panel[^}]+\}(?:.|\s)*?\{panel\}\s*/gi, '' );
+
+		// unify line breaks
 		description = description.replace( /^\s*[\r\n]/gm, '\n' );
+
+		// remove headings
+		description = description.replace( /^\s*h\d.*$/mi, '' );
+		description = description.replace( /^\*.+\*$/gm, '' );
```
[the remark this answers is not in the document]
Comment (violine1101) [reply src/mentions/SingleMention.ts:82]: These lines are usually headings, which is why I'm removing them. Though I guess there are cases where it's still important for them to stay.

File: src/mentions/SingleMention.ts [hunk 2]
```diff
@@ -70,8 +70,36 @@ export class SingleMention extends Mention {
...
+
+		// remove empty lines
+		description = description.replace( /^\s*$/gm, '' );
+
+		// remove all sections except for the first
+		description = description.replace( /\nh\d[\s\S]*$/i, '' );
+
+		// replace code block format
```
[the remark this answers is not in the document]
Comment (violine1101) [reply src/mentions/SingleMention.ts:90]: Whoops, yes, you're right

**Comment by violine1101** [comment]
After I made this PR, I came across this NPM package https://www.npmjs.com/package/jira2md that automatically converts JIRA markdown to generic markdown which should work in Discord too.

I think it's probably a better idea to rely on that package instead of reimplementing all of it here.

**Comment by NeunEinser** [comment]
> 
> 
> After I made this PR, I came across this NPM package https://www.npmjs.com/package/jira2md that automatically converts JIRA markdown to generic markdown which should work in Discord too.
> 
> I think it's probably a better idea to rely on that package instead of reimplementing all of it here.

Well, not entirely. Discord doesn't support a bunch of things general markdown does support, such as tables and headings. But it does seem like a very good starting point.

**Comment by violine1101** [comment]
I've extracted everything we need from that package into a separate util function and modified it to Discord markdown (and added multiline quotes). I think this is probably the way to go.

**Pull request merged**